src/resource-graph/azext_resourcegraph/custom.py

```diff
@@ -14,90 +14,95 @@
+        if subs_list is not None and len(subs_list) > __SUBSCRIPTION_LIMIT:
+            subs_list = subs_list[:__SUBSCRIPTION_LIMIT]
+            warning_message = "The query included more subscriptions than allowed. "\
+                              "Only the first {0} subscriptions were included for the results. "\
+                              "To use more than {0} subscriptions, "\
+                              "see the docs for examples: "\
+                              "https://aka.ms/arg-error-toomanysubs".format(__SUBSCRIPTION_LIMIT)
+            __logger.warning(warning_message)
 
+    results = []
     try:
         result_truncated = False
-        while True:
-            request_options = QueryRequestOptions(
-                top=min(first - len(results), __ROWS_PER_PAGE),
-                skip=skip + len(results),
-                skip_token=skip_token,
-                result_format=ResultFormat.object_array
-            )
-
-            request = QueryRequest(query=full_query, subscriptions=subs_list, options=request_options)
-            response = client.resources(request)  # type: QueryResponse
-            if response.result_truncated == ResultTruncated.true:
-                result_truncated = True
 
+        if first is not None:
+            first = min(first, __ROWS_PER_PAGE)
+        elif skip_token is None:
+            first = __FIRST
+
+        if skip is None and skip_token is None:
+            skip = __SKIP
+
+        request_options = QueryRequestOptions(
+            top=first,
+            skip=skip,
+            skip_token=skip_token,
+            result_format=ResultFormat.object_array,
+            allow_partial_scopes=allow_partial_scopes
+        )
+
+        request = QueryRequest(
+            query=graph_query,
+            subscriptions=subs_list,
+            management_groups=mgs_list,
+            options=request_options)
+        response = client.resources(request)  # type: QueryResponse
+        if response.result_truncated == ResultTruncated.true:
+            result_truncated = True
+
+        results.extend(response.data)
+
+        if hasattr(response, 'skip_token'):
             skip_token = response.skip_token
-            results.extend(response.data)
-
-            if len(results) >= first or skip_token is None:
-                break
 
         if result_truncated and len(results) < first:
             __logger.warning("Unable to paginate the results of the query. "
                              "Some resources may be missing from the results. "
                              "To rewrite the query and enable paging, "
                              "see the docs for an example: https://aka.ms/arg-results-truncated")
 
-    except ErrorResponseException as ex:
-        raise CLIError(json.dumps(_to_dict(ex.error), indent=4))
-
-    return results
-
+    except HttpResponseError as ex:
+        raise CLIError(json.dumps(_to_dict(ex.model.error), indent=4))
```

Comment:
Please follow the [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md) to deprecate CLIError.